pkg/domainutil/domain.go

```diff
@@ -0,0 +1,59 @@
+package domainutil
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	DefaultDomain = "astronomer.io"
+	LocalDomain   = "localhost"
+)
+
+var (
+	// CloudDomainRegex is used to differentiate cloud domain from software domain
+	// See https://github.com/astronomer/astrohub-cli/issues/7 for regexp rationale
+	// This will need to be handled as part of the permanent solution to issue #432
+	CloudDomainRegex     = regexp.MustCompile(`(cloud\.|^)astronomer(?:(-dev|-stage|-perf))?\.io`)
```

Comment:
```
(?:cloud\.|^)astronomer(?:-(dev|stage|perf))?\.io
```
this might work better. it should capture the `dev|stage|perf` part.

Reply:
this would enable you to use `matches := re.FindStringSubmatch`
if matches has a value, it's your env. if not, it's prod

Reply:
`re.MatchString` already reports if there are any matches in the string so there should not be any need to get the matches using re.FindSubMatch` and check them in any way.